util/topgen/strong_random.py

```diff
@@ -134,13 +134,15 @@ def test_runs(self):
                 i += 1
             # The runs test passes only if all numbers of runs are within
             # specified limits
-            result = (all(n_runs[0][k] >= self.RUNS_LOW_THRESHOLD[k]) and
-                      all(n_runs[1][k] >= self.RUNS_LOW_THRESHOLD[k]) and
-                      all(n_runs[0][k] <= self.RUNS_HIGH_THRESHOLD[k]) and
-                      all(n_runs[1][k] <= self.RUNS_HIGH_THRESHOLD[k])
-                      for k in range(6))
-
-            results.append(result)
+            good = True
+            for k in range(6):
+                lo = self.RUNS_LOW_THRESHOLD[k]
+                hi = self.RUNS_HIGH_THRESHOLD[k]
+                if not (lo <= n_runs[0][k] <= hi and
+                        lo <= n_runs[1][k] <= hi):
+                    good = False
+                    break
+            results.append(good)
```

Comment:
There are indeed too many lists.
If we move `good = True` outside of the outer loop `for j in range(n_seq)`, we can get rid of the the list `results` and just `return good`

Reply:
Good point: that's much cleaner! I've re-pushed with a version that I *think* does what we'd expect.